main.tf

```
@@ -165,7 +169,7 @@ resource "aws_ec2_transit_gateway_prefix_list_reference" "spokes_to_inspection_n
 # Static Route (0.0.0.0/0) from Inspection VPC to Egress VPC if:
 # 1/ Both Inspection VPC and Egress VPC are created, and the traffic inspection is "all" or "north-south".
 resource "aws_ec2_transit_gateway_route" "inspection_to_egress_default_route" {
-  count = local.inspection_and_egress_routes ? 1 : 0
+  count = local.inspection_and_egress_routes && try(local.associate_and_propagate_to_tgw["inspection"], true) ? 1 : 0
```

nitpick for **each of these changes** to meta arguments:

```suggestion
  count = (
    local.inspection_and_egress_routes && 
    try(local.associate_and_propagate_to_tgw["inspection"], true)
  ) ? 1 : 0
```

---

Fixed. All statements like that are better formatted now.